specification/src/chapter4.adoc

```diff
@@ -2,69 +2,897 @@
+Isolation guarantees provided to software also apply to device initiated transaction. 
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT NNN
+| IOPMP MUST be used to guarantee that devices assigned to lower privilege levels cannot access resources assigned to M-mode.
+
+| CAT_NNN
+| IOPMP and IOMMU MUST be used to enforce access rules for devices assigned to user applications or guests on a virtualized system.
+
+|===
+
+On a non-virtualized system, user devices can be managed by the OS which can enforce access rules for user applications.
+
+On a virtualized system, devices can be virtualized and assigned to guests by the hypervisor configuring MMU and IOMMU translation rules. 
+
+In either case, physical device access rules are always enforced by IOPMP, controlled by RoT FW in M mode, preventing unauthorized device access to M-mode resources.
 
+=== Debug and performance management
 
-=== Global Platforms TEE
+See xref:chapter2.adoc#_security_lifecycle[security lifecycle]. +
+See https://github.com/riscv-non-isa/riscv-external-debug-security[enhanced RISC-V external debug security]
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT NNN
+| External debug SHOULD be enabled separately for M-mode and non M-mode software.
+
+| CAT_NNN
+| External debug MUST only be enabled by HW RoT (M-mode external debug) or by FW RoT (non M-mode external debug).
+
+|===
+
+Enables recoverable external debug of non M-mode software. In turn enabling supply chain separation.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN
+| Self-hosted debug MAY be used for debug of non M-mode software.
+
+| CAT_NNN
+| Self-hosted debug MUST only be enabled by a higher privileged component.
+
+|===
+
+For example, within normal domain an S-mode OS can enable self-hosted debug for a user application. Only M-mode firmware should enable self-hosted debug for the S-mode OS.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN
+| FW RoT MAY disable self-hosted debug for all non M-mode software.
+
+|===
+
+For example, disable self-hosted debug in a production system for certification reasons.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN
+| External debug MUST only be enabled at boot (measuring) of the affected component.
+
+| CAT_NNN
+| Revealing self-hosted debug MUST only be enabled at boot (measuring) of the affected component
+
+| CAT_NNN
+| Trusted self-hosted debug MAY be enabled after boot (measuring) of the affected component, to an application specific governance process.
+
+|===
+
+Guarantees the system remains attestable.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#
+| Requirement
+
+| CAT_NNN
+| Lower privilege software MUST NOT be able to measure higher privilege software.
+
+| CAT_NNN
+| Software in one domain MUST NOT be able to measure software in a different domain, without consent.
```

Comment:
```suggestion
| Software in one domain MUST NOT be able to monitor the activity of software in a different domain, without consent managed by the higher privilege software.
```

Reply:
I will tweak the same statements throughout the document, for consistency.